### DISCUSSION TITLE
How to play Twitch's "VOD Track"

### DISCUSSION THREAD

**Discussion by piooop** [streamlink/streamlink#4468] [Q&A] [answered]
A couple of months ago obs implemented the twitch vod track option which allows the streamer to stream any audio they want (normally copyrighted music) through a separate audio track so that when the VOD is reuploaded to twitch that audio track isn't heard, is there any way to play that audio track through streamlink?

**Comment by bastimeyer** [comment, marked as answer]
- https://github.com/streamlink/streamlink/issues/3260
- https://help.twitch.tv/s/article/soundtrack-audio-configuration?language=en_US

> so that when the VOD is reuploaded to twitch that audio track isn't heard
> is there any way to play that audio track through streamlink?

You've already answered your own question.

Twitch implemented the "soundtrack" feature so that they can remove the unwanted/unlicensed audio in VODs without much hassle. This was previously impossible and they had to mute entire sections, which was a bad user experience. The new feature allows streamers to include two separate audio streams in their ingest stream, one for the live stream, and one for the VOD where the unlicensed audio is not included. This means that the audio you're looking for is simply not available in VODs, which is the entire goal of this feature.

**Reply by FortunateBliss** [reply]
are both audiotracks streamed when the streamer is live? and could I only grab the one without the music to only get streamers voice and video to be recorded with streamlink?

**Reply by bastimeyer** [reply]
> The new feature allows streamers to include **two separate audio streams in their _ingest stream_, one for the _live stream_, and one for the _VOD_ where the unlicensed audio is _not included_**. This means that the audio you're looking for is simply not available in VODs, which is the entire goal of this feature.

**Reply by galoomba1** [reply]
> > The new feature allows streamers to include **two separate audio streams in their _ingest stream_, one for the _live stream_, and one for the _VOD_ where the unlicensed audio is _not included_**. This means that the audio you're looking for is simply not available in VODs, which is the entire goal of this feature.

That is not what they were asking. They phrased it a little weirdly but it's obvious they're asking if they can get the track without music while watching live, not the other way around.

**Reply by bastimeyer** [reply]
And the answer is still the same, namely one specific audio track for the live stream, and the other specific audio track for the VOD. Whether you want the audio track meant for the VOD in the live stream or the audio track meant for the live stream in the VOD doesn't matter. Twitch simply doesn't make this available. If it was available, then their entire feature would be pointless.

If you still don't believe me, then have a look yourself at the API requests, the HLS multivariant and media playlist, and the track list of the MPEG-TS stream (or MPEG-4 container for the new "enhanced broadcast" streams).
